docs/static-dashboard.js

```diff
@@ -72,30 +73,60 @@ function dataToArray(data){
     let x = [];
     let yCases = [];
     let yCasesAvrg = [];
+    let yCasesCumulative = [];
     let yDeaths = [];
     let yDeathsAvrg = [];
+    let yDeathsCumulative = [];
 
     for(let i in data){
         x.push(String(data[i]["year"])+"-"+String(data[i]["month"])+"-"+String(data[i]["day"]));
 
         yCases.push(data[i]["cases"]);
 
         if (i < data.length - 6){
-            yCasesAvrg.push((data[parseInt(i)+6]["cases"]+data[parseInt(i)+5]["cases"]+data[parseInt(i)+4]["cases"]+data[parseInt(i)+3]["cases"]+data[parseInt(i)+2]["cases"]+data[parseInt(i)+1]["cases"]+data[i]["cases"])/7);
+            yCasesAvrg.push(computeAverage(data, i, 7, 'cases'))
         } else {
             yCasesAvrg.push(NaN);
-        }     
+        }
+
         yDeaths.push(data[i]["deaths"]);
 
         if (i < data.length - 6){
-            yDeathsAvrg.push((data[parseInt(i)+6]["deaths"]+data[parseInt(i)+5]["deaths"]+data[parseInt(i)+4]["deaths"]+data[parseInt(i)+3]["deaths"]+data[parseInt(i)+2]["deaths"]+data[parseInt(i)+1]["deaths"]+data[i]["deaths"])/7);
+            yDeathsAvrg.push(computeAverage(data, i, 7, 'deaths'));
         } else {
             yDeathsAvrg.push(NaN);
         }   
     };
-    return [x, yCases, yCasesAvrg, yDeaths, yDeathsAvrg];
+
+    var i = 0;
+    yCases.slice().reverse().forEach((c) => {
+        yCasesCumulative.push(i + c);
+        i += c;
+    });
+
+    i = 0;
+    yDeaths.slice().reverse().forEach((d) => {
+        yDeathsCumulative.push(i + d);
+        i += d
+    })
+
+    return [x, yCases, yCasesAvrg, yDeaths, yDeathsAvrg, yCasesCumulative.reverse(), yDeathsCumulative.reverse()];
 };
 
+
+function computeAverage(x, index, range, type) {
+    return x.slice(
+        parseInt(index), parseInt(index) + range - 1
+    )
+    .map(x => x[type])
+    .reduce((acc, val) => acc + val)
+    / range
+}
+
+function accumulate(a, b) {
+    return a.reduce((acc, val) => acc + val) + b
+}
+
 function plotGraph(){
     let covidData = extractCoutryData();
     let mydata = dataToArray(covidData);
```

Comment:
This function may needs rework to be more generic.

Reply:
Yes indeed, it would be handy for #5.